runtimes/php-8.0/server.php

```diff
@@ -0,0 +1,86 @@
+<?php
+$server = new Swoole\HTTP\Server("0.0.0.0", 3000);
+
+const DEFAULT_PATH = '/usr/code-start';
+const DEFAULT_FILE = 'index.php';
+
+function join_paths() {
+    $paths = array();
+
+    foreach (func_get_args() as $arg) {
+        if ($arg !== '') { $paths[] = $arg; }
+    }
+
+    return preg_replace('#/+#','/',join('/', $paths));
+}
+
+class Response {
+    function __construct($res) {
+        $this->res = $res;
+    }
+
+    function send($text, $status = 200) {
+        $this->res->status($status);
+        $this->res->end($text);
+    }
+
+    function json($json, $status = 200) {
+        $this->res->status($status);
+        $this->res->headers['Content-Type'] = 'application/json';
+        $this->res->end(json_encode($json, JSON_UNESCAPED_UNICODE|JSON_UNESCAPED_SLASHES));
+    }
+}
+
+
+$server->on("Request", function($req, $res) {
+    $body =  json_decode($req->getContent(), true);
+    $body['payload'] = \is_string($body['payload']) ? $body['payload'] : \json_encode($body['payload'], JSON_FORCE_OBJECT);
+
+    // Get internal_challenge header
+    $internal_challenge = $req->header['x-internal-challenge'];
+
+    if (empty($internal_challenge)) {
+        $res->status(401);
+        $res->end(json_encode([
+            'code' => 401,
+            'message' => 'Unauthorized',
+        ]));
+        return;
+    }
+
+    $key = getenv('INTERNAL_RUNTIME_KEY');
+
+    if ($key != $internal_challenge) {
+        $res->status(401);
+        $res->end(json_encode([
+            'code' => 401,
+            'message' => 'Unauthorized',
+        ]));
+        return;
+    }
+
+    $request = [
+        'env' => $body['env'] ?? [],
+        'headers' => $body['headers'] ?? [],
+        'payload' => $body['payload'] ?? '{}'
+    ];
+
+    $response = new Response($res);
+
+    try {
+        $userFunction = include(join_paths($body['path'] ?? DEFAULT_PATH, $body['file'] ?? DEFAULT_FILE));
```

Comment:
What is the goal of the `join_paths` function?

Reply:
Created PR with fix: https://github.com/open-runtimes/open-runtimes/pull/12
(hard to review, will only be 3 lines change when we merge this PHP runtime. Will fix this in all Node,Deno,PHP)